code/game/turfs/open/floor/misc_floor.dm

```diff
@@ -190,3 +190,122 @@
 	base_icon_state = "tcomms"
 	icon = 'icons/turf/floors/misc.dmi'
 	color = null
+
+//ship turfs
+/turf/open/floor/ship
+	name = "Ship Plating"
+	desc = "Report a bug if you see this."
+
+/turf/open/floor/ship/dirt
+	gender = PLURAL
+	name = "dirt"
+	desc = "Upon closer examination, it's still dirt."
+	icon = 'icons/turf/floors.dmi'
+	icon_state = "dirt"
+	footstep = FOOTSTEP_SAND
+	barefootstep = FOOTSTEP_SAND
+	clawfootstep = FOOTSTEP_SAND
+	heavyfootstep = FOOTSTEP_GENERIC_HEAVY
+	tiled_dirt = FALSE
+	var/ore_type = /obj/item/stack/ore/glass
+	var/turfverb = "dig up"
+	baseturfs = /turf/open/floor/plating
+
+/turf/open/floor/ship/dirt/attackby(obj/item/C, mob/user, params)
+	if((C.tool_behaviour == TOOL_SHOVEL) && params)
+		new ore_type(src, 2)
+		user.visible_message("<span class='notice'>[user] digs up [src].</span>", "<span class='notice'>You [turfverb] [src].</span>")
+		playsound(src, 'sound/effects/shovel_dig.ogg', 50, TRUE)
+		make_plating()
+	if(..())
+		return
+
+/turf/open/floor/ship/dirt/dark
+	icon_state = "greenerdirt"
+
+/turf/open/floor/grass/ship
+	name = "grass"
+	desc = "A patch of grass."
+	smoothing_flags = SMOOTH_BITMASK
+	smoothing_groups = list(SMOOTH_GROUP_TURF_OPEN, SMOOTH_GROUP_FLOOR_GRASS)
+	canSmoothWith = list(SMOOTH_GROUP_CLOSED_TURFS, SMOOTH_GROUP_FLOOR_GRASS)
+	layer = HIGH_TURF_LAYER
+	var/smooth_icon = 'icons/turf/floors/grass.dmi'
+	baseturfs = /turf/open/floor/ship/dirt
+
+/turf/open/floor/grass/ship/Initialize(mapload, inherited_virtual_z)
+	. = ..()
+	if(smoothing_flags)
+		var/matrix/translation = new
+		translation.Translate(-9, -9)
+		transform = translation
+		icon = smooth_icon
+
+/turf/open/floor/grass/ship/jungle
+	name = "jungle grass"
+	desc = "Greener on the other side."
+	icon_state = "junglegrass"
+	base_icon_state = "junglegrass"
+	baseturfs = /turf/open/floor/ship/dirt/dark
+	smooth_icon = 'icons/turf/floors/junglegrass.dmi'
+
+/turf/open/floor/plating/ship/water
```

Comment:
why is this a /plating, unlike the others, out of curiosity?

Reply:
The water is a plating because unlike the others it's a plating filled with water. It can be built upon to remove. It also makes it so that it cannot have wires built through it.